cloudformation/infrastructure/fargate-cluster.yaml

```diff
@@ -1,392 +1,407 @@
-        Type: List<AWS::EC2::Subnet::Id>
-
-    PrivateSubnets:
-        Description: Choose which subnets this ECS cluster should be deployed to
-        Type: List<AWS::EC2::Subnet::Id>
-
-    SecurityGroup:
-        Description: Select the Security Group to use for the ECS cluster hosts
-        Type: AWS::EC2::SecurityGroup::Id
-
-    LoadBalancerSecurityGroup:
-        Description: The SecurityGroup for load balancer
-        Type: AWS::EC2::SecurityGroup::Id
-
-    VpcId:
-        Description: The Id of the VPC for this cluster
-        Type: AWS::EC2::VPC::Id
-
-    ConcordiaVersion:
-        Type: String
-        Description: version of concordia docker images to pull and deploy
-        Default: latest
-
-    EnvName:
-        Type: String
-        Description: which environment to target
-        AllowedValues:
-            - 'dev'
-            - 'test'
-            - 'stage'
-            - 'prod'
-        ConstraintDescription: Must match a location for secret storage in secretsmanager
-
-    FullEnvironmentName:
-        Type: String
-        Description: Full name of deployment environment
-        AllowedValues:
-            - 'development'
-            - 'test'
-            - 'staging'
-            - 'production'
-
-    RedisAddress:
-        Type: String
-        Description: Redis endpoint address
-
-    RedisPort:
-        Type: String
-        Description: Redis endpoint port
-
-    MemcachedAddress:
-        Type: String
-        Description: memcached endpoint address
-
-    MemcachedPort:
-        Type: String
-        Description: memcached endpoint port
-
-    CanonicalHostName:
-        Type: String
-        Description: canonical host name of the application, e.g. crowd-test.loc.gov
-
-    DatabaseEndpoint:
-        Type: String
-        Description: Host name of the Postgres RDS service
-
-    S3BucketName:
-        Type: String
-        Description: name of the S3 bucket (public) where collection images will be stored
-
-    ExportS3BucketName:
-        Type: String
-        Description: name of the S3 bucket (public) where exported transcriptions will be stored
+  EnvironmentName:
+    Description: An environment name that will be prefixed to resource names
+    Type: String
+
+  PublicSubnets:
+    Description: The subnets for the load balancer
+    Type: List<AWS::EC2::Subnet::Id>
+
+  PrivateSubnets:
+    Description: Choose which subnets this ECS cluster should be deployed to
+    Type: List<AWS::EC2::Subnet::Id>
+
+  SecurityGroup:
+    Description: Select the Security Group to use for the ECS cluster hosts
+    Type: AWS::EC2::SecurityGroup::Id
+
+  LoadBalancerSecurityGroup:
+    Description: The SecurityGroup for load balancer
+    Type: AWS::EC2::SecurityGroup::Id
+
+  VpcId:
+    Description: The Id of the VPC for this cluster
+    Type: AWS::EC2::VPC::Id
+
+  ConcordiaVersion:
+    Type: String
+    Description: version of concordia docker images to pull and deploy
+    Default: latest
+
+  DjangoKeyId:
```

Comment:
I'm assuming this is intended to avoid issues where CloudFormation locks resources for long periods of time? That made me wonder whether the should not be a default and whether we could reuse one ID for both the Django and DB secrets.

Reply:
The problem is that the ARNs are generated with random five character suffix when the secrets are created using https://github.com/LibraryOfCongress/concordia/blob/master/cloudformation/create_secrets.sh - the name of the secret is specified, but the ARN is made up by AWS. I need to modify the master template so we can plug in those values at stack creation time.